engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java

```diff
@@ -847,4 +838,57 @@ else if ( TypeHelper.isAssignable( AnnotatedArrayType.class, annotatedType.getCl
 
 		return Optional.empty();
 	}
+
+	/**
+	 * The location of a type argument before it is really considered a constraint location.
+	 * <p>
+	 * It avoids initializing a constraint location if we did not find any constraints. This is especially useful in
+	 * a Java 9 environment as {@link ConstraintLocation#forProperty(Member) tries to make the {@code Member} accessible
+	 * which might not be possible (for instance for {@code java.util} classes).
+	 */
+	private interface TypeArgumentLocation {
+		ConstraintLocation toConstraintLocation();
+	}
+
+	private static class TypeArgumentExecutableParameterLocation implements TypeArgumentLocation {
+		private final Executable executable;
+
+		private final int index;
+
+		private TypeArgumentExecutableParameterLocation(Executable executable, int index) {
+			this.executable = executable;
+			this.index = index;
+		}
+
+		@Override
+		public ConstraintLocation toConstraintLocation() {
+			return ConstraintLocation.forParameter( executable, index );
+		}
+	}
+
+	private static class TypeArgumentPropertyLocation implements TypeArgumentLocation {
+		private final Member member;
+
+		private TypeArgumentPropertyLocation(Member member) {
+			this.member = member;
+		}
+
+		@Override
+		public ConstraintLocation toConstraintLocation() {
+			return ConstraintLocation.forProperty( member );
```

Comment:
No, it's the one which will be wrapped later. It's strictly equivalent to the existing code.